### PR TITLE
[Keyvault] BREAKING CHANGE: Fix #13752: az keyvault create not idempotent. Creating existing keyvault will fail.

### LINKED ISSUE #13752: az keyvault create not idempotent with access policies
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**

The "**az keyvault create**" command is not idempotent with **Access Policies**. On re-running the command the Access Policies are deleted/removed.

**To Reproduce**

Create a KeyVault with **az keyvault create**  e.g. "**_az keyvault create --location westus2 --name MyKeyVault --resource-group MyResourceGroup_**"

Assign an access policy to the KeyVault, either manually in the portal, or via another Azure CLI or PowershellAz command.

Run the "**az keyvault create --location westus2 --name MyKeyVault --resource-group MyResourceGroup**" command again

The access policy has been removed/deleted

**Expected behavior**
Leave the Access Policy alone

**Environment summary**

All environments

**Additional context**

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Description**

Fix #13752 
CLI will set default access_policy when running `az keyvault create`. That will overwrite existing settings if the given vault already exists.
After discussion with customer and keyvault server side engineer, we decided to keep align with what powershell does in such situation: Throw exception if the given `vault_name` exists.

**Testing Guide**

```
> az keyvault create -n existingvault -g testrg
The specified vault: existingvault already exists
```
**Notes:** All module tests containing `keyvault create` need to be re-recorded or changed to `KeyVaultPrepare`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Keyvault

**Review by Juliehzl — APPROVED** [review]
Approved for code owner part

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py [hunk 1]
```diff
@@ -1196,10 +1196,10 @@ def test_webapp_ssl(self, resource_group, resource_group_location):
 
 class WebappSSLImportCertTest(ScenarioTest):
     @ResourceGroupPreparer(location=WINDOWS_ASP_LOCATION_WEBAPP)
-    def test_webapp_ssl_import(self, resource_group):
+    @KeyVaultPreparer(location=WINDOWS_ASP_LOCATION_WEBAPP, name_prefix='kv-ssl-test', name_len=20)
+    def test_webapp_ssl_import(self, resource_group, key_vault):
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:1200]: `key_vault` is usually used as a single name.

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py [hunk 2]
```diff
@@ -644,6 +644,15 @@ def create_vault(cmd, client,  # pylint: disable=too-many-locals
                  no_self_perms=None,
                  tags=None,
                  no_wait=False):
+    from azure.core.exceptions import HttpResponseError
+    try:
+        vault = client.get(resource_group_name=resource_group_name, vault_name=vault_name)
+        if vault:
+            raise InvalidArgumentValueError('The specified vault: {} already exists'.format(vault_name))
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py:651]: In my perspective, forbidding `create` from being called multiple times is against the Azure API design:

https://docs.microsoft.com/en-us/azure/architecture/microservices/design/api-design#idempotent-operations

> The HTTP specification states that GET, PUT, and DELETE methods must be idempotent.

**Vaults - Create Or Update** is a PUT operation:

https://docs.microsoft.com/en-us/rest/api/keyvault/vaults/create-or-update

```
PUT https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}?api-version=2019-09-01
```

This PR introduces BREAKING CHANGE and we'd better call that out. This script works before but now it breaks:

```
az keyvault create -n myvault -g testrg
az keyvault create -n myvault -g testrg
``` 

Only my 2 cents.

File: src/azure-cli/azure/cli/command_modules/ams/tests/latest/test_ams_account_encryption_scenarios.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 class AmsEncryptionTests(ScenarioTest):
     @ResourceGroupPreparer(location='centralus')
     @StorageAccountPreparer(parameter_name='storage_account_for_create')
-    @KeyVaultPreparer(location='centralus')
+    @KeyVaultPreparer(location='centralus', additional_params='--enable-purge-protection')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/ams/tests/latest/test_ams_account_encryption_scenarios.py:12]: The refactoring of `KeyVaultPreparer` could be split into another PR.

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/hybrid_2018_03_01/test_keyvault_commands.py [hunk 4]
```diff
@@ -195,9 +194,9 @@ def test_keyvault_key(self, resource_group):
         self.kwargs['key_file'] = key_file
         self.cmd('keyvault key backup --vault-name {kv} -n {key} --file {key_file}')
         self.cmd('keyvault key delete --vault-name {kv} -n {key}')
-        time.sleep(10)
+        time.sleep(200)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/tests/hybrid_2018_03_01/test_keyvault_commands.py:197]: Sleeping for 200s makes live test time consuming. Any reason for the change? How about adding some retry logic, like the one in `role` module:

https://github.com/Azure/azure-cli/blob/11347b73999dc9b9d1c6f924fb42e3d20b34a767/src/azure-cli/azure/cli/command_modules/role/tests/util.py#L12-L26

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py (hunk 1 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:1200]: I just keep the origin definition within `KeyVaultPreparer` to make this PR change as small as possible.

File: src/azure-cli/azure/cli/command_modules/ams/tests/latest/test_ams_account_encryption_scenarios.py (hunk 3 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/ams/tests/latest/test_ams_account_encryption_scenarios.py:12]: It's part of re-recording creating keyvault in other modules.

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/hybrid_2018_03_01/test_keyvault_commands.py (hunk 4 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/keyvault/tests/hybrid_2018_03_01/test_keyvault_commands.py:197]: If the deletion has not finished, the later commands will fail.

The retry logic is more elegant, I can refine the code in another PR. For this one, I want it get merged **ASAP** since I have resolved conflicts for 3 times. Tough time😞

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py (hunk 2 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/keyvault/custom.py:651]: Sure, I'll add the breaking change notes in history

**Review by zhoxing-ms — APPROVED** [review]
Approved for code owner part

**Review by jiasli — APPROVED** [review]

**Comment by pascalnaber** [comment]
To raise an error when a keyvault already exists is a very bad choice. (just as choice to let it behave the same as powershell)
We are using the az cli for a reason above Powershell. The commands in Powershell are not idempotent.
Better would be to have a look at all other resources in azure regarding the create command.
For example the az storage account create 
You will not get an error that the storage account already exists. It's idempotent.

Please fix this create command to be truly idempotent. Just like all other azure resources that can be created with az.

**Comment by evelyn-ys** [comment]
@pascalnaber You are correct that the keyvault create command isn't idempotent. That's exactly what we argued internally.

The situation is other service creation is idempotent from service side, but keyvault creation is separated to two parts: one from CLI side to set default access policy, another from service side to create with initial settings.

Service won't cover setting default access policy logic during creation then if we want to make the whole process idempotent, client side(both CLI and PSH) will have to add idempotency logic for access policy which should be in server side. Both we and PSH team think it's inappropriate for client to ensure idempotency. That's why we take such design.

Sorry for any inconvenience and hope for you understanding!
